src/python/tests_extended/test_tensor_invalid_input.py

```diff
@@ -0,0 +1,396 @@
+
+        return method
+   
+    def test_pivot_bad_input_type(self):
+            
+        df = pd.DataFrame(data=dict(
+            colA=[1.0],
+            grainA=["one"]
+        ))
+
+        xf0 = LagLeadOperator(columns={'colA1': 'colA'},
+                           grain_columns=['grainA'], 
+                           offsets=[-2, -1, 1],
+                           horizon=4)
+        binarydata = xf0.fit_transform(df, as_binary_data_stream=True)
+        
+        xf1 = ForecastingPivot(columns_to_pivot=['colA1'])
+
+        xf1.fit(binarydata)
+        onnx_path = get_tmp_file('.onnx')
+        onnx_json_path = get_tmp_file('.onnx.json')
+        xf1.export_to_onnx(onnx_path,
+                            'com.microsoft.ml',
+                            dst_json=onnx_json_path,
+                            onnx_version='Stable')
+        sess = rt.InferenceSession(onnx_path)
+
+        with self.assertRaisesRegex(Exception, "ONNXRuntimeError"):
+            grainA = np.array(["one"]).reshape(1,1)
+            colA = np.array([1]).astype(np.double).reshape(1,1)
+            colA1 = np.array([1, 4, "6", float("NaN"), 2, 5, float("NaN"), float("NaN"), 3, float("NaN"), float("NaN"), 7]).reshape(1,3,4)
+            pred = sess.run(None, {"grainA":grainA,"colA":colA, "colA1":colA1 })
+            
+
+    def test_pivot_bad_shape(self):
+            
+        df = pd.DataFrame(data=dict(
+            colA=[1.0],
+            grainA=["one"]
+        ))
+
+        xf0 = LagLeadOperator(columns={'colA1': 'colA'},
+                           grain_columns=['grainA'], 
+                           offsets=[-2, -1, 1],
+                           horizon=4)
+        binarydata = xf0.fit_transform(df, as_binary_data_stream=True)
+        
+        xf1 = ForecastingPivot(columns_to_pivot=['colA1'])
+
+        xf1.fit(binarydata)
+        onnx_path = get_tmp_file('.onnx')
+        onnx_json_path = get_tmp_file('.onnx.json')
+        xf1.export_to_onnx(onnx_path,
+                            'com.microsoft.ml',
+                            dst_json=onnx_json_path,
+                            onnx_version='Stable')
+        sess = rt.InferenceSession(onnx_path)
+
+        with self.assertRaisesRegex(Exception, "ONNXRuntimeError"):
+            grainA = np.array(["one"]).reshape(1,1)
+            colA = np.array([1]).astype(np.double).reshape(1,1)
+            colA1 = np.array([1, 4, 6, float("NaN"), 2, 5, float("NaN"), float("NaN"), 3, float("NaN"), float("NaN"), 7]).reshape(1,2,6)
+            pred = sess.run(None, {"grainA":grainA,"colA":colA, "colA1":colA1 })
+
+
+
+for test_case_invalid_input in TEST_CASES_FOR_INVALID_INPUT:
+    test_name = 'test_%s' % test_case_invalid_input.replace('(', '_').replace(')', '').lower()
+    
+    method = TestOnnxExport.generate_test_method_for_bad(test_case_invalid_input)
+    setattr(TestOnnxExport, test_name, method)
+    
+TestOnnxExport.test_pivot_bad_input_type
+TestOnnxExport.test_pivot_bad_shape
```

Comment:
no need for that, the methods will be picked up by unit test system anyway

Reply:
Good to know! These two lines are deleted.